### PR TITLE
Backup: don't ignore uploadHost error

### LINKED ISSUE #3729: SM ignores backup `uploadHost` error
https://github.com/scylladb/scylla-enterprise/issues/3927
In the following fragment of SM backup code:
https://github.com/scylladb/scylla-manager/blob/69de42c491c577a5871861f7fd2af7d0e363d8fb/pkg/service/backup/worker_upload.go#L26-L32

we can see that `uploadHost` error is actually only used for logging and not returned from `f` function. This means that even when uploading host snapshot fails (a lot of `ERROR` logs), backup proceeds as nothing happened. This has a few consequences for a backup run that encountered an error (but didn't report it):
- SM reports task as `DONE` even though progress indicates otherwise (see [comment](https://github.com/scylladb/scylla-enterprise/issues/3927#issuecomment-1951133196))
- SM uploads manifest (even though not all files listed in manifest were uploaded) - restoring such manifest ends with a failure 
- SM removes snapshot files from Scylla nodes which makes it impossible to retry this backup (a fresh backup is required)

### LINKED ISSUE #3733: Backup: don't fail when there are no snapshot dirs
Consider the following scenario: backup stage upload failed on 1 host and fully succeeded on remaining hosts. Backup is retried, but it fails on all previously succeeded hosts because of this part of the code:
https://github.com/scylladb/scylla-manager/blob/69de42c491c577a5871861f7fd2af7d0e363d8fb/pkg/service/backup/worker_index.go#L160-L162
It's perfectly legal to have no snapshot dirs on resumed backup (assuming they were fully uploaded and deleted during previous run).

I'm hesitating whether just removing this check is fine or does it require more planned action. Note that this check was added because of #2426, but the description of this issue is lacking context.

### PR DESCRIPTION
**Pull request by Michal-Leszczynski**
Fixes #3729
Fixes #3733

### CODE REVIEW

**Comment by Michal-Leszczynski** [comment]
I will try to add tests later today.

**Comment by Michal-Leszczynski** [comment]
This should be included in the 3.2.6 release.

**Comment by karol-kokoszka** [comment]
Please add the test and change the PR from draft to regular one. 
I'll include it into 3.2.6 then.

**Comment by Michal-Leszczynski** [comment]
@karol-kokoszka I added a test and it turns out that it detected another, not connected issue #3733.
When commenting out validation mentioned in this issue, this test failed before fa7dccf3b6f55ff357538044b4e46ebc09d0fddc and passes after it (but it always fails when the validation is left as it is).

File: pkg/service/backup/worker_snapshot.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 	"github.com/scylladb/scylla-manager/v3/pkg/util/timeutc"
 )
 
-func (w *worker) Snapshot(ctx context.Context, hosts []hostInfo, limits []DCLimit) (err error) {
+func (w *worker) Snapshot(ctx context.Context, hosts []hostInfo, limits []DCLimit) (err error) { // nolint: dupl
```
Comment (karol-kokoszka) [inline pkg/service/backup/worker_snapshot.go:14]: you lazy bastard...
But frankly it's much better than customizing the insides

**Review by karol-kokoszka — APPROVED** [review]
Just one nit (it's about test code).
Other than that :+1:

File: pkg/service/backup/worker_snapshot.go (hunk 1 above)
Comment (Michal-Leszczynski) [reply pkg/service/backup/worker_snapshot.go:14]: I just wanted to include a quick fix so that it can be included in the release!

Comment (Michal-Leszczynski) [reply pkg/service/backup/worker_snapshot.go]: I fixed it in 3e9a9de3.

**Comment by Michal-Leszczynski** [comment]
@karol-kokoszka I decided to add a fix for failing tests in this PR. I also adressed the `// nolint: dupl` comment. Could you take a look at this PR one more time?

**Review by karol-kokoszka — APPROVED** [review]
:+1:

**Pull request merged**